web/react/components/login.jsx

```diff
@@ -173,6 +173,22 @@ export default class Login extends React.Component {
             );
         }
 
+        if (loginMessage.length > 0 && global.window.mm_config.EnableLdap === 'true' || emailSignup && global.window.mm_config.EnableLdap === 'true' || global.window.mm_config.EnableSignInWithUsername === 'true' && global.window.mm_config.EnableLdap === 'true') {
```

Comment:
This can be simplified to:

`global.window.mm_config.EnableLdap === 'true' && (loginMessage.length > 0 || emailSignup || global.window.mm_config.EnableSignInWithUsername === 'true')`

Also maybe add a couple consts like so:

`const ldapEnabled = global.window.mm_config.EnableLdap === 'true'`
`const usernameSigninEnabled = global.window.mm_config.EnableSignInWithUsername === 'true'`

So that the condition in the if statement becomes:

`ldapEnabled && (loginMessage.length > 0 || emailSignup || usernameSigninEnabled)`

Which is much more readable.